build.gradle

```diff
@@ -1,30 +1,67 @@
 apply plugin: "cpp"
 
+def boost_home = System.getenv("BOOST_HOME")
 model {
-    binaries {
-      all {
-          linker.args "-lpthread"
+  binaries {
+    all {
+      if (buildType == buildTypes.debug) {
+        cppCompiler.args "-g"
       }
+
+      cppCompiler.args "-Wextra"
+      cppCompiler.args "-Wall"
+      cppCompiler.args "-Werror"
+
+      linker.args "-lpthread"
     }
-    buildTypes {
-        debug
-        release
+  }
+  buildTypes {
+    debug
+    release
+  }
+
+  repositories {
+    libs(PrebuiltLibraries) {
+      boost_log {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_log.a")
+        }
+      }
+      boost_thread {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_thread.a")
+        }
+      }
+      boost_system {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_system.a")
+        }
+      }
     }
-    components {
-        testrunner(NativeExecutableSpec) {
-            sources {
-                cpp {
-                    source {
-                        srcDirs += "thirdparty/gtest"
-                        srcDirs += "src"
-                        srcDirs += "test"
-                        include "**/*.cc"
-                    }
-                    exportedHeaders {
-                        srcDirs += "thirdparty/gtest"
-                    }
-                }
-            }
+  }
+
+  components {
+    testrunner(NativeExecutableSpec) {
+      sources {
+        cpp {
+          source {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
+            srcDirs += "test"
+            include "**/*.cc"
+          }
+          exportedHeaders {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
```

Comment:
@AGaliuzov, seems you're right.
By the way. seems it possible to write all paths in one line. See here https://discuss.gradle.org/t/c-plugin-exportedheaders-what-does-it-do/5041

I just don't get the idea of the exportedHeaders section. In all snippets i've seend before this section was related to the library. Exported means, how I've understood, that these headers are exported by library, and then executable just uses the lib. But here we have a single executable and still use this section.